File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {      
        // Create map
        Map<Integer, Integer> uniqueNums = new HashMap<>();

        // Loop through array adding each items to the map after checking to see if the item is already in the map
        for (int i = 0; i < nums.length; i++){
            if (!uniqueNums.containsValue(nums[i])){
                uniqueNums.put(i, nums[i]);
            }
        }

        // Return the comparison between the map length and the original array length
        return !(nums.length == uniqueNums.size());
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        unique_nums = set()
        result = -1

        for i in range(len(nums)):
            if nums[i] in unique_nums:
                result = nums[i]
            else:
                unique_nums.add(nums[i])

        return result